llvm/lib/Frontend/OpenMP/OMP.cpp

```diff
@@ -193,6 +193,20 @@ bool isCombinedConstruct(Directive D) {
   return !getLeafConstructs(D).empty() && !isCompositeConstruct(D);
 }
 
+bool canHaveIterator(Clause C) {
+  // [5.2:67:5]
+  switch (C) {
+  case OMPC_affinity:
+  case OMPC_depend:
+  case OMPC_from:
+  case OMPC_map:
+  case OMPC_to:
+    return true;
+  default:
+    return false;
+  }
+}
```

Comment:
Not for this patch, but can this be specified in OMP.td and generated?

Reply:
Yes, but that's a really specific check.  The question is whether a clause modifier can introduce new symbols, and if so, what the scope is.  Maybe this can be the generalized query, but I haven't looked into it.

Reply:
Nitpick: Maybe define in the header - it's a pretty trivial function.

Reply:
Done.